rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -115,8 +154,24 @@ void SqliteStorage::write(std::shared_ptr<const rosbag2_storage::SerializedBagMe
             "' has not been created yet! Call 'create_topic' first.");
   }
 
+// The #ifdef is for debugging only, we may remove once we've iterated over this
+#ifdef EN_TRANSACTION
+  if (!active_transaction_) {
+    activate_transaction();
+  }
+  // To work with a batched insert within a trasaction
+  no_of_inserts_++;
+#endif
+
   write_statement_->bind(message->time_stamp, topic_entry->second, message->serialized_data);
   write_statement_->execute_and_reset();
+
+#ifdef EN_TRANSACTION
+  // Just for a design perspective we'll move this to a param from command line later
+  if (no_of_inserts_ > 10000) {
```

Comment:
Can we somehow shift this to a storage option item? I feel like we should do this right away while testing this PR to get a feeling about its usability and impact.